EIPS/eip-2831.md

```diff
@@ -0,0 +1,121 @@
+---
+eip: 2831
+title: Transaction Replacement Message Type
+author: Gregory Markou (@GregTheGreek)
+discussions-to: [eth magicians](https://ethereum-magicians.org/t/eip-2831-transaction-replacement-message-type/4448)
+status: Draft
+type: Standards Track
+category: Interface
+created: 2020-07-26
+requires: 1193
+---
+
+## Summary
+
+An extension to the JavaScript Ethereum Proider API ([EIP-1193](https://eips.ethereum.org/EIPS/eip-1193)) this creates a new message type in the event a transaction replacement occurs.
+
+## Abstract
+
+Exert from EIP-1193
+> A common convention in the Ethereum web application ("dapp") ecosystem is for key management software ("wallets") to expose their API via a JavaScript object in the web page.
+This object is called "the Provider".
+
+Many ingenious developments have been made by wallet developers to improve the overall user experience while interacting with the Ethereum blockchain. One specific innovation was the tx replacement, offering users the ability to effectively cancel a previously sent transaction.
+
+Transaction replacement is not a new concept, but unfortunately causes major user experience problems for dapp developers as the replaced transaction is near impossible to track. 
+
+This EIP formalizes a way for both providers and dapp developers to track transaction replacements seamlessly.
+
+This specification does not cover how one would utilize the transaction replacement message.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC-2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+> Comments like this are non-normative.
+
+### Definitions
+
+_This section is non-normative._
+
+- Provider
+  - A JavaScript object made available to a consumer, that provides access to Ethereum by means of a Client.
+- Client
+  - An endpoint that receives Remote Procedure Call (RPC) requests from the Provider, and returns their results.
+- Wallet
+  - An end-user application that manages private keys, performs signing operations, and acts as a middleware between the Provider and the Client.
+- Remote Procedure Call (RPC)
+  - A Remote Procedure Call (RPC), is any request submitted to a Provider for some procedure that is to be processed by a Provider, its Wallet, or its Client.
```

Comment:
These definitions are not present in the rest of the specification, so they probably shouldn't be included in the EIP.